frontend/src/state/api/teachers.ts

```diff
@@ -0,0 +1,11 @@
+import { useQuery } from '@tanstack/react-query';
+import { TeacherResourceService } from './generated';
+
+export const useTeachersQuery = (schoolId: string) => {
```

Comment:
```suggestion
export const useGetTeachersForSchoolQuery = (schoolId: string) => {
```
`useTeachersQuery` is a bit to generic